tests/aws/services/stepfunctions/v2/scenarios/test_base_scenarios.py

```diff
@@ -579,6 +579,94 @@ def test_map_state_config_distributed_item_selector(
             exec_input,
         )
 
+    @markers.aws.validated
+    def test_map_state_legacy_reentrant(
+        self,
+        aws_client,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        sfn_snapshot,
+    ):
+        template = ST.load_sfn_template(ST.MAP_STATE_LEGACY_REENTRANT)
+        definition = json.dumps(template)
+
+        exec_input = json.dumps({})
+        create_and_record_execution(
+            aws_client.stepfunctions,
+            create_iam_role_for_sfn,
+            create_state_machine,
+            sfn_snapshot,
+            definition,
+            exec_input,
+        )
+
+    @markers.aws.validated
+    def test_map_state_config_distributed_reentrant(
+        self,
+        aws_client,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        sfn_snapshot,
+    ):
+        # Replace MapRunArns with fixed values to circumvent random ordering issues.
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..mapRunArn", replacement="map_run_arn", replace_reference=False
+            )
+        )
+
+        template = ST.load_sfn_template(ST.MAP_STATE_CONFIG_DISTRIBUTED_REENTRANT)
+        definition = json.dumps(template)
+
+        exec_input = json.dumps({})
+        create_and_record_execution(
+            aws_client.stepfunctions,
+            create_iam_role_for_sfn,
+            create_state_machine,
+            sfn_snapshot,
+            definition,
+            exec_input,
+        )
+
+    @markers.aws.validated
+    def test_map_state_config_distributed_reentrant_lambda(
+        self,
+        aws_client,
+        create_iam_role_for_sfn,
+        create_state_machine,
+        create_lambda_function,
+        sfn_snapshot,
+    ):
+        # Replace MapRunArns with fixed values to circumvent random ordering issues.
+        sfn_snapshot.add_transformer(
+            JsonpathTransformer(
+                jsonpath="$..mapRunArn", replacement="map_run_arn", replace_reference=False
```

Comment:
Did you manually check (e.g., using `SNAPSHOT_RAW=1`) that we return the correct ARN in the right format here?

Reply:
I can confirm we do produce the correct ARN format, and we do have tests that validate this aspect of MapRuns. However, here I decided to skip this check to avoid crowding the test state with ordering reordering logic. We can however still validate that for each map input the correct output (and events) are computed.

Reply:
Reasonable. No need to double-check if we have other tests covering this part 💯